### DISCUSSION TITLE
How to use decode function?

### DISCUSSION THREAD

**Discussion by Gaurav-Harman** [lf-edge/ekuiper#2158] [Q&A] [answered]
I am trying to decode a json as below
{
  "msg": "aGVsbG8="
}

which should return me 
{
  "msg": "hello"
}

but decode function is returning 
{
  "msg": "aGVsbG8="
}

**Comment by ngjaying** [comment, marked as answer]
The decode function(https://ekuiper.org/docs/en/latest/sqls/functions/transform_functions.html#decode) now is only used to decode base64 string to 
 bytea. It is not used to decode a json string. It looks like you need `cast(decode(msg, "base64"), "string")`
